Crash when receiving webhook via MS Teams channel

**Describe the bug**
We have our PSA (Syncro https://syncromsp.com/) send notifications to MS Teams via webhook. When these notifications are received, matterbridge crashes. We have not tried other webhooks, but regular messages seem to work fine.

**To Reproduce**
Obviously not something everyone can do as the webhook post is coming from a paid service

**Expected behavior**
We are expecting the notification to be synchronized to Mattermost

**Screenshots/debug logs**
<details><summary>Debug logs</summary>

```
sudo /opt/matterbridge/matterbridge -conf /etc/matterbridge/bridge.toml -debug
[0000]  INFO main:         Enabling debug logging.
[0000]  INFO main:         Running version 1.17.4 6c442e23
[0000]  INFO router:       Parsing gateway gw
[0000]  INFO router:       Starting bridge: mattermost.mymattermost 
[0000]  INFO mattermost:   Connecting using login/password (sending and receiving)
[0000]  INFO mattermost:   Connecting bridge (team: our-team) on mm.domain.tld:443
[0000]  INFO matterclient: Found version 5.21.0.5.21.0.7257b40f883e103cbd9d764a3d679c4e.false
[0000]  INFO matterclient: found 5 users in team our-team
[0000]  INFO mattermost:   Connection succeeded
[0000]  INFO mattermost:   mattermost.mymattermost: joining tickets (ID: ticketsmattermost.mymattermost)
[0000]  INFO router:       Starting bridge: msteams.teams 
[0000] DEBUG mattermost:   Choosing login/password based receiving
[0001]  INFO msteams:      Connection succeeded
[0001]  INFO msteams:      msteams.teams: joining 19:XXXXXX@thread.tacv2 (ID: 19:XXXXXX@thread.tacv2msteams.teams)
[0001]  INFO main:         Gateway(s) started succesfully. Now relaying messages
[0001] DEBUG msteams:      getting initial messages
[0001] DEBUG msteams:      got 3 messages
[0006] DEBUG msteams:      polling for messages
[0006] DEBUG msteams:      got 3 messages
[0011] DEBUG msteams:      got 3 messages
[0017] DEBUG msteams:      got 3 messages
[0022] DEBUG msteams:      got 3 messages
[0028] DEBUG msteams:      got 3 messages
[0033] DEBUG msteams:      got 4 messages
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12c2117]

goroutine 45 [running]:
github.com/42wim/matterbridge/bridge/msteams.(*Bmsteams).poll(0xc00036d260, 0xc0001130b0, 0x30, 0x0, 0x0)
        /go/src/github.com/42wim/matterbridge/bridge/msteams/msteams.go:161 +0x417
github.com/42wim/matterbridge/bridge/msteams.(*Bmsteams).JoinChannel.func1(0xc00036d260, 0xc0001130b0, 0x30)
        /go/src/github.com/42wim/matterbridge/bridge/msteams/msteams.go:76 +0x61
created by github.com/42wim/matterbridge/bridge/msteams.(*Bmsteams).JoinChannel
        /go/src/github.com/42wim/matterbridge/bridge/msteams/msteams.go:74 +0x53
```

</details>

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Matterbridge version: /opt/matterbridge/matterbridge -version
version: 1.17.4 6c442e23

**Additional context**
bridge.toml:

```toml
[msteams.teams]
TenantID="xxx"
ClientID="xxx"
TeamID="xxx"
RemoteNickFormat="[{PROTOCOL}] <{NICK}> "

[mattermost.mymattermost]
#The mattermost hostname. (do not prefix it with http or https)
Server="mm.domain.tld:443"

#the team name as can be seen in the mattermost webinterface URL
#in lowercase, without spaces
Team="our-team"

#login/pass of your bot.
#Use a dedicated user for this and not your own!
Login="bridge"
Password="xxx"

RemoteNickFormat="[{PROTOCOL}] <{NICK}> "
PrefixMessagesWithNick=true

[[gateway]]
name="gw"
enable=true

[[gateway.inout]]
account = "mattermost.mymattermost"
channel = "tickets"

[[gateway.inout]]
account="msteams.teams"
channel="xxxxxxxx@thread.tacv2"
```

--- a/bridge/msteams/msteams.go
+++ b/bridge/msteams/msteams.go
@@ -10,6 +10,7 @@ import (
 
 	"github.com/42wim/matterbridge/bridge"
 	"github.com/42wim/matterbridge/bridge/config"
+	"github.com/davecgh/go-spew/spew"
 
 	"github.com/mattn/godown"
 	msgraph "github.com/yaegashi/msgraph.go/beta"
@@ -158,11 +159,22 @@ func (b *Bmsteams) poll(channelName string) error {
 					continue
 				}
 			}
+
+			if b.GetBool("debug") {
+				b.Log.Debug("Msg dump: ", spew.Sdump(msg))
+			}
+
+			// skip non-user message for now.
+			if msg.From.User == nil {
+				continue
+			}
+
 			if *msg.From.User.ID == b.botID {
 				b.Log.Debug("skipping own message")
 				msgmap[*msg.ID] = *msg.CreatedDateTime
 				continue
 			}
+
 			msgmap[*msg.ID] = *msg.CreatedDateTime
 			if msg.LastModifiedDateTime != nil {
 				msgmap[*msg.ID] = *msg.LastModifiedDateTime

--- a/go.mod
+++ b/go.mod
@@ -7,6 +7,7 @@ require (
 	github.com/Philipp15b/go-steam v1.0.1-0.20190816133340-b04c5a83c1c0
 	github.com/Rhymen/go-whatsapp v0.1.1-0.20200421062035-31e8111ac334
 	github.com/d5/tengo/v2 v2.4.2
+	github.com/davecgh/go-spew v1.1.1
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/go-telegram-bot-api/telegram-bot-api v4.6.5-0.20181225215658-ec221ba9ea45+incompatible
 	github.com/gomarkdown/markdown v0.0.0-20200127000047-1813ea067497

